interface/forms/soap/C_FormSOAP.class.php

```diff
@@ -38,6 +40,13 @@ function default_action()
         return $this->fetch($this->template_dir . $this->template_mod . "_new.html");
     }
 
+    function hasBilling()
+    {
+        $sql = "SELECT id FROM billing WHERE encounter = ? AND activity = 1";
+        $billingEntered = sqlQuery($sql, [$_SESSION['encounter']]);
+        return $billingEntered['id'];
```

Comment:
recommend doing ``` return $billingEntered['id'] ?? null;``` otherwise this will throw an array index error if there is no billing.